wpimath/src/main/java/edu/wpi/first/math/estimator/SwerveDrivePoseEstimator.java

```diff
@@ -153,7 +158,30 @@ public void resetPosition(
    * @return The estimated robot pose in meters.
    */
   public Pose2d getEstimatedPosition() {
-    return m_odometry.getPoseMeters();
+    return m_poseEstimate;
+  }
+
+  /**
+   * Gets the estimated robot pose at timestampSeconds.
+   *
+   * @param timestampSeconds The timestamp of the pose estimate in seconds. Note that if you don't
+   *     use your own time source by calling {@link
+   *     SwerveDrivePoseEstimator#updateWithTime(double,Rotation2d,SwerveModulePosition[])} then you
+   *     must use a timestamp with an epoch since FPGA startup (i.e., the epoch of this timestamp is
+   *     the same epoch as {@link edu.wpi.first.wpilibj.Timer#getFPGATimestamp()}.) This means that
+   *     you should use {@link edu.wpi.first.wpilibj.Timer#getFPGATimestamp()} as your time source
+   *     or sync the epochs.
+   * @return The estimated robot pose in meters.
+   */
+  public Pose2d getEstimatedPosition(double timestampSeconds) {
+    if (m_poseBuffer.getInternalBuffer().isEmpty()) {
+      return m_poseEstimate;
+    }
+    // current to old odometry delta
```

Comment:
Please capitalize the first letter of comments. There's other instances of this too.